### DISCUSSION TITLE
How to organize public assets for each blog post?

### DISCUSSION THREAD

**Discussion by apoorvmote** [vercel/next.js#15090] [Help] [answered]
Currently all images in `public` folder is served from `/` path. 
Now I have 3 blog posts blog1, blog2, blog3 each of them have 3 different images but all 3 images are called feature.jpg 
Now Imagine I have 30 or 300 or 3000 blogposts. My job to to make sure each image name is not repeated on used on other blog posts.

**Comment by cristiand391** [comment, marked as answer]
You can create multiple sub-folders in `public` and access them from `/`.

**Reply by apoorvmote** [reply]
Yes I tested this and it did work. But I am getting finicky results. Many times image from subfolders is not showing up. I have to refresh localhost:3000 many times until image finally shows up. 

I have few more kind of related questions.

**Is it better to write `amp` pages with `jsx` or is it better with `.md`?**

The benefit of using `.md` is the front matter(date or cateogory) that I can use to sort blog post and generate blog list pages. 

**How would someone go about automatically generating list pages from `amp` page written in `jsx`?**

**How good is `.md` to `amp` page conversion?** 
I am using `marked` to convert `.md` to `html`, adding dangerous inner html and making `export config = { amp: true }` and hoping that nextjs will do the rest. Like importing relevant `amp` libraries.

**Reply by cristiand391** [reply]
I haven't used amp with next.js so I can't answer those questions, sorry.

Regarding to assets not loading, if you are frequently experiencing problems you should open a [new issue](https://github.com/vercel/next.js/issues).

**Reply by apoorvmote** [reply]
Anyway thanks for the help. Only reason I picked up Nextjs is because it has best amp integration and amp people recommended it.
